internal/seclang/rule_parser.go

```diff
@@ -452,21 +450,45 @@ actionLoop:
 			ckey.WriteRune(c)
 		}
 		if i+1 == len(actions) {
-			f, err := actionsmod.Get(ckey.String())
+			// last action, returned disruptiveActionIndex is not needed
+			res, _, err = appendRuleAction(res, &ckey, &cval, disruptiveActionIndex)
 			if err != nil {
 				return nil, err
 			}
-			res = append(res, ruleAction{
-				Key:   ckey.String(),
-				Value: cval.String(),
-				F:     f,
-				Atype: f.Type(),
-			})
 		}
 	}
 	return res, nil
 }
 
+func appendRuleAction(res []ruleAction, ckey *strings.Builder, cval *strings.Builder, disruptiveActionIndex int) ([]ruleAction, int, error) {
+	f, err := actionsmod.Get(ckey.String())
+	if err != nil {
+		return res, unset, err
+	}
+	if f.Type() == rules.ActionTypeDisruptive && disruptiveActionIndex != unset {
+		// There can only be one disruptive action per rule (if there are multiple disruptive
+		// actions present, or inherited, only the last one will take effect).
+		// Therefore, if we encounter another disruptive action, we replace the previous one.
+		res[disruptiveActionIndex] = ruleAction{
+			Key:   ckey.String(),
+			Value: cval.String(),
+			F:     f,
+			Atype: f.Type(),
+		}
+	} else {
+		if f.Type() == rules.ActionTypeDisruptive {
+			disruptiveActionIndex = len(res)
```

Comment:
Can we have the parser ensure this is added at the end if that's what's expected?

Reply:
If my interpretation of `in the end` (equal to `at the end of the chain`) is correct, we are good as we are. the loop will iterate and evaluate both the disruptive action and flow actions, regardless of the order